lib/tasks/accounts/update_adopter_status.rake

```diff
@@ -4,33 +4,33 @@ namespace :accounts do
   task update_adopter_status: [:environment] do
     last_id = nil
     loop do
-      contacts = OpenStax::Salesforce::Remote::Contact.select(
-        :id,
-        :adoption_status,
-        :accounts_uuid
-      )
-      .where("Accounts_UUID__c != null")
-      .order(:Id)
-      .limit(250)
-      contacts = contacts.where("Id > '#{last_id}'") unless last_id.nil?
-      contacts = contacts.to_a
-      last_id = contacts.last&.id
+      users = User.where(adopter_status: nil).order(:id).limit(250)
+      users = users.where("id > #{last_id}") unless last_id.nil?
```

Comment:
With the above change you shouldn't need this anymore (or to track last_id). You just find 250 of them and update so adopter_status is not nil and they won't be found in later iterations.